src/main/java/com/google/sps/servlets/DestinationDataServlet.java

```diff
@@ -0,0 +1,152 @@
+  private static final String HINT3_PARAMETER = "hint3";
+  private static final String OBSCURITY_PARAMETER = "obscurity";
+  private static final String TAG_PARAMETER = "tag";
+  private static final String REDIRECT_URL = "/destination-data";
+
+  // Temporarily stores the destination created by the user
+  public List<Destination> destinations = new ArrayList<>();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter(NAME_PARAMETER);
+
+    LatLng location =
+        new LatLng.Builder()
+            .withLat(Double.parseDouble(request.getParameter(LAT_PARAMETER)))
+            .withLng(Double.parseDouble(request.getParameter(LNG_PARAMETER)))
+            .build();
+
+    String city = request.getParameter(CITY_PARAMETER);
+    String description = request.getParameter(DESCRIPTION_PARAMETER);
+
+    Riddle riddle =
+        new Riddle.Builder()
+            .withPuzzle(request.getParameter(RIDDLE_PARAMETER))
+            .withHint(request.getParameter(HINT1_PARAMETER))
+            .withHint(request.getParameter(HINT2_PARAMETER))
+            .withHint(request.getParameter(HINT3_PARAMETER))
+            .build();
+
+    /* Retrieves the obscurity level chosen by the user as a List of Strings and converts the level to an Enum Obscurity value */
+    String obscureLevel = request.getParameter(OBSCURITY_PARAMETER);
+    Destination.Obscurity level = convertLevelToEnum(obscureLevel);
+
+    /* Retrieves the tags selected by the user as a List of Strings and converts them into a Set of Enum Tags */
+    List<String> tags =
+        Arrays.stream(request.getParameterValues(TAG_PARAMETER))
+            .filter(tag -> tag != null)
+            .collect(Collectors.toList());
+    Set<Destination.Tag> checkedTags = convertTagsToEnum(tags);
+
+    Destination destination =
+        new Destination.Builder()
+            .withName(name)
+            .withLocation(location)
+            .withCity(city)
+            .withDescription(description)
+            .withRiddle(riddle)
+            .withTags(checkedTags)
+            .withObscurity(level)
+            .build();
+
+    /* TODO: Store the data in datastore rather than locally */
+    destinations.add(destination);
+    /* TODO: redirect back index.html once the data is stored in datastore */
+    response.sendRedirect(REDIRECT_URL);
+  }
+
+  /* Retrieves the most recent destination created by the user, turns it into a JSON formatted string,
+   *and displays the JSON-ified String on /destination-data
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+    String json = gson.toJson(destinations.get(destinations.size() - 1));
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  public Set<Destination.Tag> convertTagsToEnum(List<String> tags) {
```

Comment:
this method should be private, not public, since it's not used by any other classes